agentai/conversation.py

```diff
@@ -20,19 +22,42 @@ def __init__(self, history: List[Message] = [], id: Optional[str] = None):
             "function": "magenta",
         }
         self.id = id
+        self.max_tokens = max_tokens
 
-    def add_message(self, role: str, content: str, name: Optional[str] = None):
+    def add_message(self, role: str, content: str, name: Optional[str] = None) -> None:
         message_dict = {"role": role, "content": content}
-        if name is not None:
+        if name:
             message_dict["name"] = name
         message = Message(**message_dict)
         self.history.append(message)
 
-    def display_conversation(self):
+    def display_conversation(self) -> None:
         for message in self.history:
             print(
                 colored(
                     f"{message.role}: {message.content}\n\n",
-                    self.role_to_color[message.role],
+                    self.role_to_color[message.role.lower()],
                 )
             )
+
+    def get_history(self) -> List[Message]:
+        """Function to get the conversation history based on the number of tokens"""
+        local = threading.local()
+        try:
+            enc = local.gpt2enc
+        except AttributeError:
+            enc = tiktoken.get_encoding("gpt2")
+            local.gpt2enc = enc
```

Comment:
Check and confirm if the GPT4 tokeniser is same as gpt2? From what I recall, this is wrong. The tokeniser depends on the LLM.